pkg/task/storage.go

```diff
@@ -230,6 +230,29 @@ func (s *Storage) rangeIter(prefix string, start time.Time, end time.Time) (task
 	return tasks, nil
 }
 
+// AllTasks returns all tasks in the store
+func (s *Storage) AllTasks() (tasks []*Task, err error) {
+	rng := util.Range{
+		Start: nil, Limit: nil,
+	}
+
+	tasks = make([]*Task, 0)
+
+	iter := s.db.NewIterator(&rng, nil)
+	defer iter.Release()
+
+	for iter.Next() {
+		tsk := &Task{}
+
+		err := json.Unmarshal(iter.Value(), tsk)
+		if err != nil {
+			return nil, err
+		}
+		tasks = append(tasks, tsk)
+	}
+	return tasks, nil
+}
+
```

Comment:
Do you know if this will contain ALL the tasks the cluster ran since its creation?
Nothing to fix now, but if that's the case we should plan for starting time issues on long-lived clusters.

Reply:
Yes, that's correct - it will return all tasks present in the DB. However, I don't see this slowing down the startup time even when there are thousands of runs, unless the DB driver is very inefficient.
In either case, if we notice performance issues, we can refine this to retrieve the tasks from the past X days, instead of all tasks.